modules/transfer-connectors/main.tf

```diff
@@ -0,0 +1,336 @@
+data "external" "connector_ips" {
+  program = ["bash", "-c", <<-EOF
+    # Try to describe the connector and extract any IP information
+    connector_info=$(aws transfer describe-connector --connector-id "${aws_transfer_connector.sftp_connector.id}" --region "${data.aws_region.current.id}" 2>/dev/null || echo "{}")
+    
+    # Extract IP addresses if they exist and format as comma-separated string
+    ips=$(echo "$connector_info" | jq -r '.Connector.ServiceManagedEgressIpAddresses // [] | join(",")')
+    
+    if [ -n "$ips" ] && [ "$ips" != "" ]; then
+      echo "{\"ips\": \"$ips\", \"status\": \"found\"}"
+    else
+      echo "{\"ips\": \"\", \"status\": \"not_available\", \"note\": \"No IP addresses available via API\"}"
+    fi
+  EOF
+  ]
+  
+  depends_on = [aws_transfer_connector.sftp_connector]
+}
+
+
+
+# IAM role for Lambda
+resource "aws_iam_role" "lambda_role" {
+  count = local.should_scan ? 1 : 0
+  
+  name = "transfer-connector-lambda-role-${var.connector_name}"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+      }
+    ]
+  })
+
+  tags = var.tags
+}
+
+# IAM policy for Lambda
+resource "aws_iam_role_policy" "lambda_policy" {
+  count = local.should_scan ? 1 : 0
+  
+  name = "transfer-connector-lambda-policy-${var.connector_name}"
+  role = aws_iam_role.lambda_role[0].id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = "arn:aws:logs:*:*:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "transfer:DescribeConnector"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+#####################################################################################
+# SFTP Connector
+#####################################################################################
+
+resource "aws_transfer_connector" "sftp_connector" {
+  depends_on = [
+    aws_iam_role_policy_attachment.connector_policy_attachment,
+    aws_iam_role_policy_attachment.connector_logging_policy
+  ]
+  
+  access_role = aws_iam_role.connector_role.arn
+  url         = var.url
+
+  # SFTP config without trusted_host_keys (optional)
+  dynamic "sftp_config" {
+    for_each = local.effective_secret_id != null ? [1] : []
+    content {
+      user_secret_id = local.effective_secret_id
+    }
+  }
+
+  logging_role = local.logging_role
+  security_policy_name = var.security_policy_name
+
+  tags = merge(
+    var.tags,
+    {
+      Name = var.connector_name
+    }
+  )
+
+}
+
+resource "null_resource" "discover_and_test_connector" {
+  depends_on = [aws_transfer_connector.sftp_connector]
+  
+  provisioner "local-exec" {
```

Comment:
make the local exec conditional with a bool attribute, add comments and a section in the readme describing what this will do along with the dependencies required to test connectors